scipy/stats/_censored_data.py

```diff
@@ -0,0 +1,192 @@
+import numpy as np
+
+
+def _validate_lower_upper(lower, upper):
+    if lower.ndim != 1 or upper.ndim != 1:
+        raise ValueError('The input arrays must be one-dimensional.')
+    if np.shape(lower) != np.shape(upper):
+        raise ValueError('lower and upper must have the same length.')
+    if np.isnan(lower).any() or np.isnan(upper).any():
+        raise ValueError('lower and upper must not contain nan.')
+    if (np.isinf(lower) & np.isinf(upper)).any():
+        raise ValueError('lower and upper must not both be infinite.')
+    if (lower > upper).any():
+        raise ValueError('Elements of lower must not be greater than the '
+                         'corresponding elements of upper.')
+
+
+def _validate_x_censored(x, censored):
+    x = np.asarray(x)
+    if x.ndim != 1:
+        raise ValueError('x must be one-dimensional.')
+    censored = np.asarray(censored)
+    if censored.ndim != 1:
+        raise ValueError('censored must be one-dimensional')
+    if (~np.isfinite(x)).any():
+        raise ValueError('x must not contain nan or inf.')
+    if censored.size != x.size:
+        raise ValueError('x and censored must have the same length.')
+    return x, censored.astype(bool)
+
+
+class CensoredData:
+    """
+    Instances of this class represent censored data.
+
+    Instances may be passed to the ``fit`` method of continuous
+    univariate SciPy distributions for MLE fitting.
+
+    Left-, right-, and interval-censored data can be represented.
+
+    For interval-censored data, pass the lower and upper limits of
+    the intervals as one-dimensional arrays to the ``CensoredData``
+    constuctor.  By using the values `np.inf` and `-np.inf`, left-
+    and right-censored data may also be created with the constructor.
+
+    For convenience, the class methods ``left_censored`` and
+    ``right_censored`` are provided to create a ``CensoredData``
+    instance from a single one-dimensional array of observations
+    and a corresponding boolean array to indicate which observations
+    are censored.
+
+    Examples
+    --------
+    In the most general case, a censored data set may contain values that
+    are left-censored, right-censored, interval-censored and not censored.
+    For example, here we create a data set with length five:
+
+    >>> from scipy.stats import CensoredData
+    >>> data = CensoredData([-np.inf, 1, 1.5, 2, 10],
+    ...                     [0,       1, 1.5, 3, np.inf])
+    >>> print(data)
+    CensoredData(5 values: 2 not censored, 1 left-censored,
+    1 right-censored, 1 interval-censored)
+
+    The first value is left-censored; the true (but unknown) value is
+    less than or equal to 0.  The second and third values, 1 and 1.5,
+    are not censored.  The fourth value is interval-censored; the true
+    value is in the interval [2, 3].  The last value is right-censored;
+    the true value is greater than or equal to 10.
```

Comment:
This description is confusing to me. There actually aren't any values yet, the constructor only sets lower and upper bounds. If you say "the value is right-censored" I'd interpret that as there actually being a data value and it being above the true interval. That's not the case here.

Reply:
See the new Parameters sections for clarification.  I also copy-edited the text a bit--I hope it helps.